### DISCUSSION TITLE
Why do variable types get lost when using lambda expressions and dependency injection?

### DISCUSSION THREAD

**Discussion by mht2953658596** [fastapi/fastapi#11766] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/", description="test")
async def test(x: int, y: int = Depends(lambda x: x)):
    print(type(y))
    return {"x": x, "y": y}
This will print the str type.


when
@router.get("/", description="test")
async def test(x: int, y: int = Depends(lambda x: x+1)):
    return {"x": x, "y": y}

raise an error：
async def test(x: int, y: int = Depends(lambda x: x + 1)):
TypeError: can only concatenate str (not "int") to str
```

### Description

When one query parameter depends on another, the type of y becomes str.

But when I use a regular function instead of a lambda expression, the type does not convert to str. Why is that?

### Operating System

Windows

### FastAPI Version

0.95.2

### Pydantic Version

1.10.8

### Python Version

Python 3.8.16

**Comment by YuriiMotov** [comment, marked as answer]
That's beause FastAPI inspects the signature of the function you wrap by `Depends` to get the type of parameters that function requires.
And when you use a lambda function, FastAPI can't find function's parameter type annotations and uses default type (`str`).
